### PR TITLE
Atomic link

### PR DESCRIPTION
**Pull request by zkochan**
This is a pretty good structural change to node_modules. However, it will work only with preserve-symlinks = false. Which shouldn't be an issue

This change also separates bundled dependencies from regular ones, which solves the issues of linking bundled dependencies from the global store

Closes: #546, #553

### CODE REVIEW

File: src/api/extendOptions.ts [hunk 1]
```diff
@@ -41,5 +41,6 @@ export default (opts?: PnpmOptions): StrictPnpmOptions => {
   if (extendedOpts.flatTree === true && !extendedOpts.preserveSymlinks) {
     throw new Error('`--preserve-symlinks` and so `--flat-tree` are not supported on your system, make sure you are running on Node ≽ 6.3.0')
   }
+  extendedOpts.preserveSymlinks = false // TODO: remove preserve-symlinks related code
```
Comment (iamstarkov) [inline src/api/extendOptions.ts]: why do you keep it?

Comment (iamstarkov) [reply src/api/extendOptions.ts]: i mean if everything should work fine with normal nodejs, why do we need this line of code?

Comment (zkochan) [reply src/api/extendOptions.ts]: I'll remove that option

**Comment by zkochan** [comment]
I noticed today a new issue with preserve-syminks, when used with karma. So I believe it makes sense to deliver this one even without solution to [this issue](https://github.com/pnpm/pnpm/issues/548#issuecomment-274903241)

**Pull request merged**